frontends/precinct-scanner/src/screens/scan_warning_screen.test.tsx

```diff
@@ -62,6 +74,40 @@ test('overvote', () => {
   expect(fetchMock.done()).toBe(true);
 });
 
+test('overvote when casting overvotes is disallowed', () => {
+  mockOf(isFeatureFlagEnabled).mockImplementation(
+    (flag: EnvironmentFlagName) => {
+      return flag === EnvironmentFlagName.DISALLOW_CASTING_OVERVOTES;
+    }
+  );
+
+  const contest = electionSampleDefinition.election.contests.find(
+    (c): c is CandidateContest => c.type === 'candidate'
+  )!;
+
+  renderScreen({
+    adjudicationReasonInfo: [
+      {
+        type: AdjudicationReason.Overvote,
+        contestId: contest.id,
+        optionIds: contest.candidates.map(({ id }) => id),
+        optionIndexes: contest.candidates.map((c, i) => i),
+        expected: 1,
+      },
+    ],
+  });
+
+  screen.getByRole('heading', { name: 'Too Many Votes' });
+  screen.getByText(
+    new RegExp(
+      `There are too many votes marked in the contest for: ${contest.title}.`
+    )
+  );
+  expect(
+    screen.queryAllByRole('button', { name: 'Cast Ballot As Is' })
+  ).toStrictEqual([]);
```

Comment:
I often use [.not.toBeInTheDocument()](https://testing-library.com/docs/guide-disappearance/#nottobeinthedocument) for these cases

```ts
expect(
    screen.queryByRole('button', { name: 'Cast Ballot As Is' })
  ).not.toBeInTheDocument()
```